### PR TITLE
Add Checker Interface for PrivacyChecker and DependencyChecker

### PR DESCRIPTION
**Pull request by doodzik**
## What are you trying to accomplish?

When I was going through the codebase I noticed that PrivacyChecker and DependencyChecker conform to an interface but don't have one defined. In this PR I'm adding an interface to codify that interface. This will make it easier to add additional sorbet signatures and makes sure that the contract for the checker classes is enforced.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Code refactor (non-breaking change that doesn't add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] It is safe to simply rollback this change.

### CODE REVIEW

File: lib/packwerk/dependency_checker.rb [hunk 1]
```diff
@@ -2,9 +2,12 @@
 # frozen_string_literal: true
 
 require "packwerk/violation_type"
+require "packwerk/checker_interface"
 
 module Packwerk
   class DependencyChecker
+    require "packwerk/checker_interface"
```
Comment (tomstuart) [inline lib/packwerk/dependency_checker.rb:9]:
```suggestion
    include CheckerInterface
```

**Comment by tomstuart** [comment]
Ditto [my comment on #39](https://github.com/Shopify/packwerk/pull/39#issuecomment-715157905) about commits and interface names.

I think it’s a good idea to add an interface for `PrivacyChecker` and `DependencyChecker` 🎉

**Comment by exterm** [comment]
I agree - this refactor is a good idea, and we should try to align on a way to solve the problem that interface implementation looks like module inclusion and it's confusing.

I added a comment to https://github.com/Shopify/packwerk/pull/39 as well.

**Comment by doodzik** [comment]
I made some adjustments to the PR and split the commits up. Please have a look again :)

File: lib/packwerk/checker.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# typed: true
+# frozen_string_literal: true
+
+require "sorbet-runtime"
+require "packwerk/reference_lister"
+
+module Packwerk
+  module Checker
```
Comment (tomstuart) [inline lib/packwerk/checker.rb:8]: This interface is a good idea, and `Checker` looks like a good name based on my current understanding of our naming conventions 👍

**Review by tomstuart — APPROVED** [review]
Good to go, as long as you convince yourself that the `lib/packwerk.rb` change is right 👍

File: lib/packwerk.rb [hunk 3]
```diff
@@ -34,7 +34,6 @@
 require "packwerk/parsers"
 require "packwerk/privacy_checker"
 require "packwerk/reference_extractor"
-require "packwerk/reference_lister"
```
[the remark this answers is not in the document]
Comment (doodzik) [reply lib/packwerk.rb:37]: I think it makes sense, but if we decide that we don't want to do it I will change it back 👍

**Pull request merged**